src/main/java/de/tum/cit/aet/artemis/core/web/PasskeyResource.java

```diff
@@ -0,0 +1,101 @@
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import de.tum.cit.aet.artemis.core.config.Constants;
+import de.tum.cit.aet.artemis.core.domain.User;
+import de.tum.cit.aet.artemis.core.domain.converter.BytesConverter;
+import de.tum.cit.aet.artemis.core.dto.PasskeyDto;
+import de.tum.cit.aet.artemis.core.repository.UserRepository;
+import de.tum.cit.aet.artemis.core.repository.webauthn.ArtemisUserCredentialRepository;
+import de.tum.cit.aet.artemis.core.security.annotations.EnforceAtLeastStudent;
+
+/**
+ * REST controller for public endpoints regarding the webauthn (Web Authentication) API, e.g. used for passkeys.
+ */
+@Profile(PROFILE_CORE)
+@RestController
+@RequestMapping("api/core/passkey/")
+public class PasskeyResource {
+
+    public static final String ENTITY_NAME = "passkey";
+
+    private static final Logger log = LoggerFactory.getLogger(PasskeyResource.class);
+
+    private final UserRepository userRepository;
+
+    private final ArtemisUserCredentialRepository artemisUserCredentialRepository;
+
+    @Value("${artemis.user-management.passkey.enabled:false}")
+    public boolean enabled;
+
+    /**
+     * REST controller for managing passkeys.
+     * Provides endpoints for retrieving and deleting passkeys associated with a user.
+     *
+     * This controller is only active when the "core" profile is enabled.
+     *
+     * @param userRepository                  for accessing user data
+     * @param artemisUserCredentialRepository for managing user credentials
+     */
+    public PasskeyResource(UserRepository userRepository, ArtemisUserCredentialRepository artemisUserCredentialRepository) {
+        this.userRepository = userRepository;
+        this.artemisUserCredentialRepository = artemisUserCredentialRepository;
+    }
+
+    /**
+     * Checks if the passkey feature is enabled.
+     *
+     * @throws NotAllowedException if the passkey feature is not enabled
+     */
+    private void checkIfPasskeyFeatureIsEnabled() throws NotAllowedException {
+        if (!enabled) {
+            log.info("If you want to enable the passkey feature, please set the property '{}' to true in your application properties.", Constants.PASSKEY_ENABLED_PROPERTY_NAME);
+            throw new NotAllowedException("Passkey feature is not enabled");
+        }
+    }
+
+    @GetMapping("user")
+    @EnforceAtLeastStudent
+    public ResponseEntity<List<PasskeyDto>> getPasskeys() {
+        checkIfPasskeyFeatureIsEnabled();
+
+        User user = userRepository.getUser();
+        log.info("Retrieving passkeys for user with id: {}", user.getId());
+
+        List<PasskeyDto> passkeys = artemisUserCredentialRepository.findPasskeyDtosByUserId(BytesConverter.longToBytes(user.getId()));
+
+        return ResponseEntity.ok(passkeys);
+    }
+
+    /**
+     * @param credentialIdBase64Encoded as base64 encoded url string
+     */
```

Comment:
Please turn this into a proper JavaDoc

Reply:
/**
     * Deletes a passkey associated with the given credential ID.
     *
     * This endpoint allows users to delete a specific passkey by providing its
     * Base64-encoded credential ID. The passkey feature must be enabled for this
     * operation to succeed.
     *
     * @param credentialIdBase64Encoded of the passkey to delete
     * @return a {@link ResponseEntity} with HTTP status 204 (No Content) if the deletion is successful
     * @throws NotAllowedException if the passkey feature is disabled
     */
     
Do you think that this javadoc is better?

From my perspective it does not really add value there as this should be the case for every delete endpoint, shouldn't it? 😅

Reply:
I think generally a JavaDoc should have at least a short description in addition to the @ tags.
Usually we just do something like `* DELETE /guided-tour-settings/:settingsKey : delete guided tour setting of the current user`, as that at least has the use of finding endpoints easier via Ctrl+F search.

Reply:
![image](https://github.com/user-attachments/assets/27f23afd-729d-4ad6-8062-bb3bca88e992)

Not quite sure which one of the plugins creates the view or if it is even IntelliJ built in 😅